Mage.Sets/src/mage/cards/s/SilverScrutiny.java

```diff
@@ -0,0 +1,154 @@
+package mage.cards.s;
+
+import mage.MageObject;
+import mage.abilities.SpellAbility;
+import mage.abilities.costs.mana.VariableManaCost;
+import mage.abilities.dynamicvalue.common.ManacostVariableValue;
+import mage.abilities.effects.common.DrawCardSourceControllerEffect;
+import mage.abilities.effects.common.InfoEffect;
+import mage.abilities.keyword.FlashAbility;
+import mage.cards.AdventureCardSpell;
+import mage.cards.Card;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.AsThoughEffectType;
+import mage.constants.CardType;
+import mage.constants.WatcherScope;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.game.stack.Spell;
+import mage.game.stack.SpellStack;
+import mage.game.stack.StackObject;
+import mage.watchers.Watcher;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.UUID;
+
+/**
+ *
+ * @author weirddan455
+ */
+public final class SilverScrutiny extends CardImpl {
```

Comment:
Please add some tests for this class given the amount of custom code needed for it.

Reply:
Yea, I need to look into how to assert that you want a test to fail (ex. cast on opponent's end step with X=5 or something).

Reply:
It may not be possible without changes to the test suite.

I few tests I've had to write inside a try {} catch block